src/hooks/use-cart.ts

```diff
@@ -8,10 +9,27 @@ import { useEcomAPI } from '~/api/ecom-api-context-provider';
 
 export const useCart = () => {
     const ecomAPI = useEcomAPI();
+    const [updatingCartItems, setUpdatingCartItems] = useState<string[]>([]);
+
     const { data: cartData } = useCartData();
-    const { data: cartTotals } = useCartTotals();
-    const { trigger: updateItemQuantity } = useUpdateCartItemQuantity();
-    const { trigger: removeItem } = useRemoveItemFromCart();
+    const { data: cartTotalsData, isUpdatingOnCartChange: isCartTotalsUpdatingOnCartChange } =
+        useCartTotals();
+    const cartTotals = cartTotalsData?.data;
+
+    const { trigger: triggerUpdateItemQuantity } = useUpdateCartItemQuantity();
+    const { trigger: triggerRemoveItem } = useRemoveItemFromCart();
+
+    const updateItemQuantity = async ({ id, quantity }: { id: string; quantity: number }) => {
+        setUpdatingCartItems((prev) => [...prev, id]);
+        await triggerUpdateItemQuantity({ id, quantity });
+        setUpdatingCartItems((prev) => prev.filter((itemId) => itemId !== id));
```

Comment:
It's super incovinient to do it like this. Can't we move hooks to the cart item component and use `updateCartItemQuantity.isLoading`?

Reply:
- The idea was to move cart logic into one hook to use easily between similar projects.
- `CartItem` is view component

Reply:
But if because of that we need to write something like that, I'm against it

Reply:
I think it's a bit better.

```
    const updateItemQuantity = ({ id, quantity }: { id: string; quantity: number }) => {
        setUpdatingCartItems((prev) => [...prev, id]);
        triggerUpdateItemQuantity(
            { id, quantity },
            {
                onSuccess: () =>
                    setUpdatingCartItems((prev) => prev.filter((itemId) => itemId !== id)),
            },
        );
    };
```

Reply:
It's the same

Reply:
@Letushev I'm with you :) but don't forget that it's a template for Codux.. and for Codux users it's good to have as many dumb components as possible so they are easy to work with in boards.

Reply:
to be honest when I refactored the code in ecom-starter template (extracting view components) I wasn't aware of (and didn't think of) this updating items UI. Refactoring made sense then.. now I agree that it's more on the edge but still believe that having pure view components is highly beneficial in our context.

@alisey what do you think?

Reply:
I don't want to get deep into this, but `setUpdatingCartItems` doesn't look too bad. "Here's a list of items currently being updated". Makes sense. If it's hidden inside the hook, and the user doesn't need to worry about, it's a win for me.